jerry-core/ecma/base/ecma-globals.h

```diff
@@ -996,6 +996,11 @@ typedef double ecma_number_t;
 #define ECMA_MAX_CHARS_IN_STRINGIFIED_NUMBER 64
 
 /**
+ * Maximum number of digits in string representation of ecma-number exponent part
+ */
+#define ECMA_MAX_NUMBER_OF_DIGITS_OF_EXPONENT 21
```

Comment:
If we introduce such a constant then it should be used throughout the codebase. I've grepped for 21 under jerry-core/ecma and found other occurrences:

ecma-builtin-number-prototype.c:
L585 (ecma_builtin_number_prototype_object_to_fixed)
L727 (ecma_builtin_number_prototype_object_to_exponential)
L915 (ecma_builtin_number_prototype_object_to_precision)

ecma-helpers-conversion.c:
L1134-1138 (ecma_number_to_utf8_string)
L1146-1150 (ecma_number_to_utf8_string)

It should be thoroughly checked of course whether these 21s are the same 21 that we try to replace. But if they are, the new macro should be used in place of them.

Reply:
After some discussion https://www.ecma-international.org/ecma-262/5.1/#sec-15.7.4.5 does not provide precise explanation about these 'magic numbers', so we can keep them instead of defining a new macro for them.

Reply:
@rerobika For the sake of clarity: so, none of the above 5 instances of the numeric literal 21 have any connection with the "maximum number of digits in string representation of ecma-number exponent part" concept? But the functions already touched in the current patch (`ecma_double_to_binary_floating_point` and `ecma_builtin_number_prototype_object_to_fixed`) do use this concept? Seems strange to me.

OK, just checked ES5.1 15.7.4.5 (Number.prototype.toFixed) step 7 (currently touched). No explanation there either. So, the patch, as is now, introduces a new macro and inconsistently uses it in the code base. I start to get the feeling that we don't need this macro at all. Magic constants in the code are evil for sure, but such inconsistencies are even worse, IMHO.

(BTW, it seems to me that the current macro name + docstring are misleading. There is no way that we have 21 digits in the exponent. In double precision floating point, exponent is 11 bits, which is 4 base-10 digits at most.)

Summary: I think we should revert to NOT introducing a new macro and NOT touching other parts of the code base but where the bug is fixed.

Reply:
Yeah, it seems like a better idea.